### PR TITLE
feat(TDP-3114/backend): limit the size of locally uploaded dataset

### PR DESCRIPTION
**Pull request by bdiouf**
* Add Unitary test
* Add a new parameter in the datasets application.properties file

**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-XXXX

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/770/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/770/)

File: dataprep-backend-service/src/main/java/org/talend/dataprep/dataset/store/content/StrictlyBoundedInputStream.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.talend.dataprep.exception.TDPException;
+import org.talend.dataprep.exception.error.DataSetErrorCodes;
+
+/**
+ * A bounded input stream that throws a {@link TDPException} when the maximum input stream size is exceeded.
+ */
+public class StrictlyBoundedInputStream extends InputStream {
```
Comment (hmdebenque) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/dataset/store/content/StrictlyBoundedInputStream.java]: Is it not a copy of org.apache.commons.io.input.BoundedInputStream but using an exception to express it has hit the limit?

File: dataprep-dataset/src/main/java/org/talend/dataprep/dataset/service/DataSetService.java [hunk 2]
```diff
@@ -308,7 +312,7 @@ public String create(
 
             // Save data set content
             LOG.debug(marker, "Storing content...");
-            contentStore.storeAsRaw(dataSetMetadata, content);
+            contentStore.storeAsRaw(dataSetMetadata, new StrictlyBoundedInputStream(content, maximumInputStreamSize));
```
Comment (hmdebenque) [inline dataprep-dataset/src/main/java/org/talend/dataprep/dataset/service/DataSetService.java]: Are we sure that underlying resources are cleaned if the limit is reached?
We wouldn't want to keep incomplete amazon objects or partially written files. Even for local files, there does not seem to be any cleaning if write fails.

File: dataprep-backend-service/src/main/java/org/talend/dataprep/dataset/store/content/StrictlyBoundedInputStream.java (hunk 1 above)
Comment (bdiouf) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/dataset/store/content/StrictlyBoundedInputStream.java]: no it is not the same concept, the org.apache.commons.io.input.BoundedInputStream abruptly closes the input stream if it reaches the limit and you will not know if the backed inputstream has been closed normally or not.

File: dataprep-dataset/src/main/java/org/talend/dataprep/dataset/service/DataSetService.java (hunk 2 above)
Comment (bdiouf) [inline dataprep-dataset/src/main/java/org/talend/dataprep/dataset/service/DataSetService.java]: Actually there is a small problem with the whole process while importing dataset (local, s3, ...).

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/818/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/818/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/821/) [:x:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/821/)

**Pull request merged**